drivers/blk/sdmmc/src/sddf_helper.c

```diff
@@ -0,0 +1,43 @@
+#include <sddf/blk/queue.h>
+#include <blk_config.h>
+#include <stdint.h>
+
+blk_queue_handle_t queue_handle_memory;
+blk_queue_handle_t *queue_handle = &queue_handle_memory;
+
+blk_req_queue_t *blk_req_queue;
+blk_resp_queue_t *blk_resp_queue;
+
+blk_storage_info_t *blk_config;
+
+void blk_queue_init_helper() {
+    blk_queue_init(queue_handle, blk_req_queue, blk_resp_queue, BLK_QUEUE_SIZE_DRIV);
+    blk_config->sector_size = 512;
+    blk_config->block_size = 1;
+    blk_config->capacity = 0xFFFFFFFFFF;
+    blk_config->ready = true;
+}
+
+uint8_t blk_queue_empty_req_helper() {
+    return blk_queue_empty_req(queue_handle);
+}
+
+uint8_t blk_queue_full_resp_helper() {
+    return blk_queue_full_resp(queue_handle);
+}
+
+uint8_t blk_enqueue_resp_helper(uint8_t status, uint16_t success, uint32_t id) {
+    // It would be better if we do not use int but use int8_t
+    if (blk_enqueue_resp(queue_handle, status, success, id) == 0) {
```

Comment:
It's most likely because bindgen can't convert the header and so we have to have some wrapper functions.

I'm not too much of a fan of this though, we should just be able to have an object that only includes `<sddf/blk/queue.h>` etc for the Rust bindings.

Reply:
I have not had much luck with header-only APIs and bindgen in the past.

@Cheng-Li1 did you try bindgen before adding these wrappers?

Reply:
No, I admit I have never used this before. But I guess if I use this bindgen, Ineed to keep track of the queue_handle in the Rust code?